### ISSUE TITLE
'wcstombs': This function or variable may be unsafe

### ISSUE THREAD

**Issue by sbobko** [KhronosGroup/OpenCL-ICD-Loader#93]
Hello,
I tried to build ID_loader on Windows ver. 1903 (SDK 10.0.18362.0 and WDK 10.0.18362.1)
and got the following error:
`OpenCL-ICD-Loader\windows\icd_windows_dxgk.c(142): error C4996: 'wcstombs': This function or variable may be unsafe. Consider using wcstombs_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.`
Can you help me solve this problem?

**Comment by bashbaug** [comment]
Hello Sergey,

I don't think I've seen this warning since I usually build the ICD loader without WDK support.

A few possible ways to fix it:

1. Switch to `wcstombs_s`, as suggested by the warning.
2. Silence the warning by #defining `_CRT_SECURE_NO_WARNINGS`.  This could be done by adding a line to the CMake file:

```
if (WIN32)
    target_link_libraries (OpenCL cfgmgr32.lib)
    # add this line:
    target_compile_definitions (OpenCL PRIVATE _CRT_SECURE_NO_WARNINGS)
```

@nikhiljnv, any other suggestions?

Thanks!

**Comment by sbobko** [comment]
Thanks for the quick response. 
This will really help, but I would like to solve the problem without editing the code locally. if it is necessary I can correct it myself and submit the merge request.

**Comment by nikhiljnv** [comment]
I would prefer using secured version of API to avoid any security risks.

**Comment by sbobko** [comment]
I'm also for a secure API :) Is it worth waiting for updates?

**Comment by bashbaug** [comment]
I believe this issue will be fixed by #102 - thanks!
